### PR TITLE
[#10103] Add Remind All button for "Participants who have not responded to any question" tab

### LINKED ISSUE #10103: Add Remind All button for "Participants who have not responded to any question" tab
V6 screenshot:
image

This will pop-up:
image

Hint:
- You need to reuse the component `SendRemindersToStudentModalComponent`
- You need to interact with the endpoint `RemindFeedbackSessionSubmissionAction` by using the method `remindFeedbackSessionSubmissionForStudent`.
- There are already example usages in the codebase (Note that `Select all respondents` should be selected when the model pops out (The behavior is not the same as that of the existing use case))

### PR DESCRIPTION
**Pull request by Derek-Hardy**
Fixes #10103 

**Snapshot**:
remind-all-button

**Outline of Solution**
- add `Remind All` button in `instructor-session-no-response-panel`
- add function `openSendReminderModal` to handle `send-reminders-to-student-modal` logic
- add in-line CSS to position the button
- improve UI spacing on `send-reminders-to-student-modal`
- fix comment typo in .ts file of `send-reminders-to-student-modal`

### CODE REVIEW

File: src/web/app/pages-instructor/instructor-session-result-page/instructor-session-no-response-panel.component.ts [hunk 1]
```diff
@@ -57,6 +77,52 @@ export class InstructorSessionNoResponsePanelComponent implements OnInit, OnChan
     }
   }
 
+  openSendReminderModal(event: any): void {
```
Comment (xpdavid) [inline src/web/app/pages-instructor/instructor-session-result-page/instructor-session-no-response-panel.component.ts:80]: Let's make the request sent from the root page.

File: src/web/app/pages-instructor/instructor-session-result-page/instructor-session-no-response-panel.component.ts [hunk 2]
```diff
@@ -57,6 +77,52 @@ export class InstructorSessionNoResponsePanelComponent implements OnInit, OnChan
...
+    event.stopPropagation();
+
+    const courseId: string = this.session.courseId;
+    const feedbackSessionName: string = this.session.feedbackSessionName;
+
+    const modalRef: NgbModalRef = this.modalService.open(SendRemindersToStudentModalComponent);
+    modalRef.componentInstance.courseId = courseId;
+    modalRef.componentInstance.feedbackSessionName = feedbackSessionName;
+
+    forkJoin(
```
Comment (xpdavid) [inline src/web/app/pages-instructor/instructor-session-result-page/instructor-session-no-response-panel.component.ts:90]: If we sent the request at the root page, can we reuse `noResponseStudents` and don't send the request again?

https://github.com/TEAMMATES/teammates/blob/55bc31496aa06ffd31350593cc7b7258af408af0/src/web/app/pages-instructor/instructor-session-result-page/instructor-session-result-page.component.ts#L192-L213

File: src/web/app/pages-instructor/instructor-session-result-page/instructor-session-no-response-panel.component.ts (hunk 1 above)
Comment (xpdavid) [reply src/web/app/pages-instructor/instructor-session-result-page/instructor-session-no-response-panel.component.ts:78]: I should have said `InstructorSessionResultPageComponent` (root page)

Comment (Derek-Hardy) [reply src/web/app/pages-instructor/instructor-session-result-page/instructor-session-no-response-panel.component.ts:72]: All REST requests are moved to the root page.

File: src/web/app/pages-instructor/instructor-session-result-page/instructor-session-no-response-panel.component.ts [hunk 3]
```diff
@@ -57,6 +69,34 @@ export class InstructorSessionNoResponsePanelComponent implements OnInit, OnChan
...
+    modalRef.componentInstance.feedbackSessionName = feedbackSessionName;
+    modalRef.componentInstance.studentListInfoTableRowModels
+      = this.allStudents.map((student: Student) => ({
+        email: student.email,
+        name: student.name,
+        teamName: student.teamName,
+        sectionName: student.sectionName,
+
+        hasSubmittedSession: !giverSet.has(student.email),
+        isSelected: true,
```
Comment (xpdavid) [inline src/web/app/pages-instructor/instructor-session-result-page/instructor-session-no-response-panel.component.ts:92]: I suppose only those who do not answer the session should be selected?

File: src/web/app/pages-instructor/instructor-session-result-page/instructor-session-no-response-panel.component.ts [hunk 4]
```diff
@@ -57,6 +69,34 @@ export class InstructorSessionNoResponsePanelComponent implements OnInit, OnChan
     }
   }
 
+  openSendReminderModal(event: any): void {
+    event.stopPropagation();
+
+    const courseId: string = this.session.courseId;
+    const feedbackSessionName: string = this.session.feedbackSessionName;
+
+    const giverEmails: string[] = this.noResponseStudents.map((student: Student) => student.email);
```
Comment (xpdavid) [inline src/web/app/pages-instructor/instructor-session-result-page/instructor-session-no-response-panel.component.ts:78]: nonResponseStudentEmails?

File: src/web/app/pages-instructor/instructor-session-result-page/instructor-session-no-response-panel.component.ts [hunk 5]
```diff
@@ -57,6 +69,34 @@ export class InstructorSessionNoResponsePanelComponent implements OnInit, OnChan
...
+    const giverSet: Set<string> = new Set(giverEmails);
```
Comment (xpdavid) [inline src/web/app/pages-instructor/instructor-session-result-page/instructor-session-no-response-panel.component.ts:79]: nonResponseStudentEmailSet?

File: src/web/app/pages-instructor/instructor-session-result-page/instructor-session-no-response-panel.component.ts [hunk 6]
```diff
@@ -38,7 +46,11 @@ export class InstructorSessionNoResponsePanelComponent implements OnInit, OnChan
   isTabExpanded: boolean = false;
 
   noResponseStudentsInSection: Student[] = [];
-  constructor() { }
+
+  @Output() studentsToRemindData: EventEmitter<StudentListInfoTableRowModel[]> = new EventEmitter();
```
Comment (xpdavid) [inline src/web/app/pages-instructor/instructor-session-result-page/instructor-session-no-response-panel.component.ts:50]:
```suggestion
  @Output() studentsToRemindEvent: EventEmitter<StudentListInfoTableRowModel[]> = new EventEmitter();
```

File: src/web/app/pages-instructor/instructor-session-result-page/instructor-session-no-response-panel.component.ts (hunk 3 above)
Comment (xpdavid) [reply src/web/app/pages-instructor/instructor-session-result-page/instructor-session-no-response-panel.component.ts:92]: Apology for the misleading hint :P

Comment (Derek-Hardy) [reply src/web/app/pages-instructor/instructor-session-result-page/instructor-session-no-response-panel.component.ts:92]: Alright, in that case `select all respondents` at the top is not selected by default. Only those with `Submitted?` status as **No** will be selected by default. Is it the intended behaviour?

Comment (xpdavid) [reply src/web/app/pages-instructor/instructor-session-result-page/instructor-session-no-response-panel.component.ts:92]: Yup. By clicking button, you just want to send reminder to all nonResponseStudents.

**Pull request merged**